src/Entity/Subscription.php

```diff
@@ -0,0 +1,315 @@
+use Drupal\apigee_m10n\Entity\Property\StartDatePropertyAwareDecoratorTrait;
+use Drupal\Core\Entity\EntityTypeInterface;
+
+/**
+ * Defines the Subscription entity class.
+ *
+ * @\Drupal\apigee_edge\Annotation\EdgeEntityType(
+ *   id             = "subscription",
+ *   label          = @Translation("Subscription"),
+ *   label_singular = @Translation("Subscription"),
+ *   label_plural   = @Translation("Subscriptions"),
+ *   label_count = @PluralTranslation(
+ *     singular = "@count Subscription",
+ *     plural   = "@count Subscriptions",
+ *   ),
+ *   handlers = {
+ *     "storage"             = "Drupal\apigee_m10n\Entity\Storage\SubscriptionStorage",
+ *     "access"              = "Drupal\apigee_edge\Entity\EdgeEntityAccessControlHandler",
+ *     "permission_provider" = "Drupal\apigee_edge\Entity\EdgeEntityPermissionProviderBase",
+ *     "list_builder"        = "Drupal\apigee_m10n\Entity\ListBuilder\SubscriptionListBuilder",
+ *     "form" = {
+ *       "default"     = "Drupal\apigee_m10n\Entity\Form\SubscriptionForm",
+ *       "unsubscribe" = "Drupal\apigee_m10n\Entity\Form\UnsubscribeConfirmForm",
+ *     },
+ *   },
+ *   links = {
+ *     "collection-by-developer" = "/user/{user}/monetization/subscriptions",
+ *     "unsubscribe-form"        = "/user/{user}/monetization/subscription/{subscription}/unsubscribe",
+ *   },
+ *   entity_keys = {
+ *     "id" = "id",
+ *   },
+ *   permission_granularity = "entity_type",
+ *   admin_permission       = "administer subscription",
+ *   field_ui_base_route    = "apigee_m10n.settings.subscription",
+ * )
+ */
+class Subscription extends FieldableEdgeEntityBase implements SubscriptionInterface {
+
+  use EndDatePropertyAwareDecoratorTrait;
+  use StartDatePropertyAwareDecoratorTrait;
+
+  public const ENTITY_TYPE_ID = 'subscription';
+
+  /**
+   * The rate plan this subscription belongs to.
+   *
+   * @var \Drupal\apigee_m10n\Entity\RatePlanInterface
+   */
+  protected $rate_plan;
+
+  /**
+   * Constructs a `subscription` entity.
+   *
+   * @param array $values
+   *   An array of values to set, keyed by property name.
+   * @param null|string $entity_type
+   *   Type of the entity. It is optional because constructor sets its default
+   *   value.
+   * @param \Apigee\Edge\Entity\EntityInterface|null $decorated
+   *   The SDK entity that this Drupal entity decorates.
+   *
+   * @throws \ReflectionException
+   */
+  public function __construct(array $values, ?string $entity_type = NULL, ?EdgeEntityInterface $decorated = NULL) {
+    /** @var \Apigee\Edge\Api\Management\Entity\DeveloperAppInterface $decorated */
+    $entity_type = $entity_type ?? static::ENTITY_TYPE_ID;
+    parent::__construct($values, $entity_type, $decorated);
+    // Save entity references in this class as well as the decorated instance.
+    if (!empty($values['ratePlan']) && $values['ratePlan'] instanceof DrupalRatePlanInterface) {
+      $this->setRatePlan($values['ratePlan']);
```

Comment:
In order to let entity references work correctly, we return the Drupal `rate_plan` entity for `getRatePlan()` so we need to save it somewhere. There is `RatePlan::createFrom($sdk_rate_plan)` so maybe this could use some improvement. I'll add a TODO to look at this.